src/policy_evaluator.rs

```diff
@@ -16,12 +16,76 @@ use crate::cluster_context::ClusterContext;
 use crate::policy::Policy;
 use crate::validation_response::ValidationResponse;
 
+#[derive(Clone, PartialEq)]
 pub enum PolicyExecutionMode {
     KubewardenWapc,
     Opa,
     OpaGatekeeper,
 }
 
+impl serde::ser::Serialize for PolicyExecutionMode {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::ser::Serializer,
+    {
+        let mode = match self {
+            PolicyExecutionMode::KubewardenWapc => "kubewarden",
+            PolicyExecutionMode::Opa => "opa",
+            PolicyExecutionMode::OpaGatekeeper => "gatekeeper",
+        };
+        serializer.serialize_str(mode)
+    }
+}
+
+struct StringVisitor;
+
+impl<'de> serde::de::Visitor<'de> for StringVisitor {
+    type Value = String;
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        write!(formatter, "a string")
+    }
+
+    fn visit_str<E>(self, s: &str) -> Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        Ok(s.to_owned())
+    }
+}
+
+impl<'de> serde::de::Deserialize<'de> for PolicyExecutionMode {
+    fn deserialize<D>(deserializer: D) -> Result<PolicyExecutionMode, D::Error>
+    where
+        D: serde::de::Deserializer<'de>,
+    {
+        let mode = deserializer.deserialize_str(StringVisitor)?;
+        match mode.as_str() {
+            "kubewarden" => Ok(PolicyExecutionMode::KubewardenWapc),
+            "opa" => Ok(PolicyExecutionMode::Opa),
+            "gatekeeper" => Ok(PolicyExecutionMode::OpaGatekeeper),
+            _ => Err(serde::de::Error::custom(format!(
+                "{} is not a valid execution mode",
+                mode
+            ))),
+        }
+    }
+}
```

Comment:
Maybe I'm missing something, but do we need to implement this? Cannot we derive it?